include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.h

```diff
@@ -313,13 +319,9 @@ class MultigridPreconditionerBase : public PreconditionerBase<Number>
   // In case of global coarsening, this is the mapping associated to the fine level triangulation.
   std::shared_ptr<dealii::Mapping<dim> const> mapping;
 
-  // Only relevant for global coarsening,where this vector contains coarse level mappings,
+  // Only relevant for global coarsening, where this vector contains coarse level mappings,
   // and the fine level mapping as the last element of the vector.
-  std::vector<std::shared_ptr<MappingDoFVector<dim, Number>>> coarse_grid_mappings;
-
-  // Only relevant for local-smoothing-multigrid and in case that a mapping of type MappingQCache is
-  // used.
-  std::shared_ptr<MappingDoFVector<dim, Number>> mapping_dof_vector;
```

Comment:
Why has this been removed?

Reply:
I think for the use in the multigrid algorithm, we only need a `dealii::Mapping`. So my goal is to make the implementation more generic. If I see it correctly, we only need `MappingDoFVector` during construction, and could then let a `std::shared_ptr<dealii::Mapping>` point to this object. Following this strategy, the "construction code" can later be shifted to a lambda function implemented outside multigrid, e.g. in the application.